x/concentrated-liquidity/position.go

```diff
@@ -380,6 +401,23 @@ func (k Keeper) validatePositionsAndGetTotalLiquidity(ctx sdk.Context, owner sdk
 			return 0, 0, 0, sdk.Dec{}, types.PositionOwnerMismatchError{PositionOwner: position.Address, Sender: owner.String()}
 		}
 
+		// Check that all the positions have no underlying lock that has not yet matured.
+		positionHasUnderlyingLock := k.positionHasUnderlyingLockInState(ctx, positionId)
+		if positionHasUnderlyingLock {
+			underlyingLockId, err := k.GetPositionIdToLock(ctx, positionId)
+			if err != nil {
+				return 0, 0, 0, sdk.Dec{}, err
+			}
+
+			lockIsMature, err := k.isLockMature(ctx, underlyingLockId)
+			if err != nil {
+				return 0, 0, 0, sdk.Dec{}, err
+			}
+			if !lockIsMature {
+				return 0, 0, 0, sdk.Dec{}, types.LockNotMatureError{PositionId: positionId, LockId: underlyingLockId}
+			}
+		}
```

Comment:
Please update godoc for this keeper

Reply:
godoc updated here https://github.com/osmosis-labs/osmosis/pull/4882/commits/aa7bbd23185ff4bc94500dbb6cc1cca7cc1c60b8